### PR TITLE
StorageSystemTables add metadata_version

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add column `metadata_version` of ReplicatedMergeTree table in `system.tables` system table.

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 0ecdc9a848523273a63f98a6fde58c1cbad32b42 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59942/0ecdc9a848523273a63f98a6fde58c1cbad32b42/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59942/0ecdc9a848523273a63f98a6fde58c1cbad32b42/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: docs/en/operations/system-tables/tables.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ Columns:
 
 - `metadata_modification_time` ([DateTime](../../sql-reference/data-types/datetime.md)) - Time of latest modification of the table metadata.
 
+- `metadata_version` ([Int32](../../sql-reference/data-types/int-uint.md)) - Metadata version for ReplicatedMergeTree table, -1 for non replicated table.
```
Comment (alexey-milovidov) [inline docs/en/operations/system-tables/tables.md:30]: Let's use 0 for non-replicated tables.

**Review by alexey-milovidov — COMMENTED** [review]
Missing tests.

File: src/Storages/System/StorageSystemTables.cpp [hunk 2]
```diff
@@ -43,6 +43,7 @@ StorageSystemTables::StorageSystemTables(const StorageID & table_id_)
         {"data_paths", std::make_shared<DataTypeArray>(std::make_shared<DataTypeString>()), "Paths to the table data in the file systems."},
         {"metadata_path", std::make_shared<DataTypeString>(), "Path to the table metadata in the file system."},
         {"metadata_modification_time", std::make_shared<DataTypeDateTime>(), "Time of latest modification of the table metadata."},
+        {"metadata_version", std::make_shared<DataTypeInt32>(), "Metadata version for ReplicatedMergeTree table."},
```
[the remark this answers is not in the document]
Comment (kitaisreal) [reply src/Storages/System/StorageSystemTables.cpp:46]: This metadata_version seems like too complex to cover in documentation without mentioning ALTERs and ZooKeeper `metadata` node version.
For example in StorageInMemoryMetadata.h there is such comment: 
```
/// Version of metadata. Managed properly by ReplicatedMergeTree only
/// (zero-initialization is important)
int32_t metadata_version = 0;
```
Generally idea of having those in `system.tables` is to provide some interface to the advanced user to understand which `metadata_version` of ZooKeeper `metadata` node is currently in ClickHouse memory.

**Review by alesapin — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**